### PR TITLE
Get cargo working on i686-pc-windows-msvc

### PR DESCRIPTION
**Pull request by alexcrichton**
These commits provide the necessary support to get cargo working in the 32-bit MSVC context. Now that we've got 32-bit MSVC nightlies rolling we can add support!

### CODE REVIEW

**Comment by alexcrichton** [comment]
r? @brson 

I'd also like to hold off on r+ until appveyor gives the go-ahead at least as I've also added 32-bit coverage to our matrix, but I haven't verified it works yet.

**Comment by alexcrichton** [comment]
Ok, I've got some more work to do to get appveyor working on 32/64 builds

File: tests/test_cargo_compile.rs [hunk 1]
```diff
@@ -864,6 +864,11 @@ Caused by:
 });
 
 test!(lto_build {
+    // FIXME: currently this hits a linker bug on 32-bit MSVC
+    if cfg!(all(target_env = "msvc", target_pointer_width = "32")) {
+        return
+    }
```
Comment (alexcrichton) [inline tests/test_cargo_compile.rs]: Note that this is connected to https://github.com/rust-lang/rust/pull/27224, so once that lands and makes its way into the nightlies this test can be re-enabled on 32-bit MSVC again.

**Comment by brson** [comment]
r=me

**Pull request merged**